### PR TITLE
[BBS-292] Release v0.1.0

### PR DESCRIPTION
**Pull request by FrancescoCasalegno**
Fixes [BBS-292](https://bbpteam.epfl.ch/project/issues/browse/BBS-292).

## Description

1. **Remove GitHub Actions workflow to automatically upload to PyPI.** This was done after discussing with @jankrepl about pros and cons of this choice. See https://github.com/BlueBrain/atlas_alignment/pull/19 for the rationale.
2. **DVC Dockerfiles use versions specified in`requirements.txt`**. This change improves the reproducibility of results with DVC. Before this, we were just using the dependencies specified in `setup.py` which means that there was no guarantee on the version of each dependency used to run the DVC pipelines and results could have slightly changed.
3. **Upgrade `prodigy`** version to fix [this error](https://support.prodi.gy/t/importerror-cannot-import-name-schema-from-pydantic/3948). 
4. **Run `dvc repro`** (and `dvc push` to update the remote storage).
5. **DVC-untrack BSV** model. This reduces the `.dvc/cache` to only `4.4 GiB`.

## How to test?
1. **Get Prodigy wheel**. 
```cp /raid/prodigy_downloads/1.10.7/prodigy-1.10.7-cp36.cp37.cp38-cp36m.cp37m.cp38-linux_x86_64.whl path/to/Search```
1. **Get images for running the `ner` and `sentemb` pipelines.** You have two options.
     1. **Option1** — You can use the images I already built on DGX1: `bbs_dvc_ner:v0.1.0` and `bbs_dvc_sentemb:v0.1.0`. In the following I assume you use this option—otherwise replace the image names in the following commands! 
     1. **Option 2** — You can build the images yourself: 
         - `docker build -f data_and_models/pipelines/ner/Dockerfile  -t <ner_image_name> .` 
         - `docker build -f data_and_models/pipelines/sentence_embedding/Dockerfile  -t <sentemb_image_name> .` 
1. **Reproduce the `ner` pipeline.**
    1. Get all dvc files `dvc pull`
    2. Run container`docker run --rm -it <your_options> --name <ner_container_name> bbs_dvc_ner:v0.1.0`
    2. Move to directory `cd data_and_models/pipelines/ner/`
    3. Reproduce pipeline `dvc repro` (or `dvc repro -f` if you have some time and want to be 100% sure)
    4. Check that nothing changed: `git diff` should be empty
1. **Reproduce the `sentemb` pipeline.**
    1. Get all dvc files `dvc pull` (if you run the `ner` pipeline already, then this step can be skipped)
    2. Run container`docker run --rm -it <your_options> --name <sentemb_container_name> bbs_dvc_sentemb:v0.1.0`
    2. Move to directory `cd data_and_models/pipelines/sentence_embedding/`
    3. Reproduce pipeline `dvc repro` (or `dvc repro -f` if you have some time and want to be 100% sure)
    4. Check that nothing changed: `git diff` should be empty

## Checklist

- [x] This PR refers to an issue from the [issue tracker](https://github.com/BlueBrain/Search/issues).
  (if it is not the case, please create an issue first).
- [x] Documentation and `whatsnew.rst` updated.
  (if needed)
- [x] All CI tests pass.

### CODE REVIEW

**Comment by pafonta** [comment]
It seems that

```
docker build -f data_and_models/pipelines/sentemb/Dockerfile -t <sentemb_image_name> .
```

should instead be

```
docker build -f data_and_models/pipelines/sentence_embedding/Dockerfile -t <sentemb_image_name> .
```

in the test instructions.

**Comment by jankrepl** [comment]
When building the NER image I get the following error on the line `RUN pip install -r requirements.txt `
```
ERROR: Cannot uninstall 'ruamel-yaml'. It is a distutils installed project and thus we cannot accurately determine which files belong to it which would lead to only a partial uninstall.
```
The internet suggests to use `--ignore-installed`, however, I am not sure how replicable this error is.

**Comment by FrancescoCasalegno** [comment]
@jankrepl I have rebuild the image right now from scratch (using a fresh clone of this repo) and I cannot reproduce the error with `RUN pip install -r requirements.txt` that you mention. Could you please give me some more detail about it? 

Maybe a fresh `git clone` could help?

**Comment by jankrepl** [comment]
> Hello @FrancescoCasalegno !
> 
> I cannot run `dvc repro` inside the container. I have tried with
> 
> ```
> /Search/data_and_models/pipelines/ner# dvc repro --pull
> ```
> 
> but I get the error
> 
> ```
> Verifying data sources in stage: '../../annotations/ner/annotations10_EmmanuelleLogette_2020-08-28_raw1_raw5_10EntityTypes.jsonl.dvc'
> ERROR: failed to reproduce '../../annotations/ner/annotations10_EmmanuelleLogette_2020-08-28_raw1_raw5_10EntityTypes.jsonl.dvc': missing data 'source': ../../annotations/ner/annotations10_EmmanuelleLogette_2020-08-28_raw1_raw5_10EntityTypes.jsonl
> ```

I got exactly the same error.

**Comment by jankrepl** [comment]
> @jankrepl I have rebuild the image right now from scratch (using a fresh clone of this repo) and I cannot reproduce the error with `RUN pip install -r requirements.txt` that you mention. Could you please give me some more detail about it?
> 
> Maybe a fresh `git clone` could help?

As discussed, I got this error when using the most recent `ontinuumio/miniconda3` with sha `7838d0ce65783b0d944c19d193e2e6232196bada9e5f3762dc7a9f07dc271179

**Comment by FrancescoCasalegno** [comment]
> As discussed, I got this error when using the most recent ontinuumio/miniconda3 with digest sha `7838d0ce65783b0d944c19d193e2e6232196bada9e5f3762dc7a9f07dc271179`

Thanks, I now also see the error! This seems to be a known issue, and reading [this SO post](https://stackoverflow.com/questions/63383400/error-cannot-uninstall-ruamel-yaml-while-creating-docker-image-for-azure-ml-a) as well as [this GH issue](https://github.com/pypa/pip/issues/5247) I think we only have 3 options:
1. Manually downgrade `pip` to `<1.10`
2. Manually `rm -rf` the `ruamel` install
3. `pip install` with `--ignore-installed`

I think the third option seems the least ugly, I am now testing if it works fine.

**Comment by jankrepl** [comment]
> > As discussed, I got this error when using the most recent ontinuumio/miniconda3 with digest sha `7838d0ce65783b0d944c19d193e2e6232196bada9e5f3762dc7a9f07dc271179`
> 
> Thanks, I now also see the error! This seems to be a known issue, and reading [this SO post](https://stackoverflow.com/questions/63383400/error-cannot-uninstall-ruamel-yaml-while-creating-docker-image-for-azure-ml-a) as well as [this GH issue](https://github.com/pypa/pip/issues/5247) I think we only have 3 options:
> 
> 1. Manually downgrade `pip` to `<1.10`
> 2. Manually `rm -rf` the `ruamel` install
> 3. `pip install` with `--ignore-installed`
> 
> I think the third option seems the least ugly, I am now testing if it works fine.

Yes, I went for 3. locally and it worked. Anyway, I think we should pin the `continuumio/miniconda3` to a specific tag in our Dockerfiles.

**Comment by FrancescoCasalegno** [comment]
@jankrepl  and @pafonta — I always did `dvc pull` + `dvc repro -f` rather than the `dvc repro --pull` that you mentioned in your comments, so I never noticed that this latter doesn't work. But you are definitely right, thanks!

It seems that [this is a known issue that should be fixed in the future](https://github.com/iterative/dvc/issues/4742). I updated the instructions to suggest a testing workflow that should run smoothly using `dvc pull` + `dvc repro -f`.

File: data_and_models/pipelines/ner/Dockerfile [hunk 1]
```diff
@@ -29,22 +29,20 @@ RUN true \
...
 # Instal BlueBrainSearach -- revision can be a branch, sha, or tag
 ARG BBS_REVISION=v0.1.0
 ADD . /src
 WORKDIR /src
-RUN pip install git+file://$PWD@$BBS_REVISION
+RUN git checkout $BBS_REVISION
+# remove ruamel-yaml: https://github.com/pypa/pip/issues/5247#issuecomment-381550610
+RUN rm -rf /opt/conda/lib/python3.8/site-packages/ruamel*
+RUN pip install $PWD
+RUN pip install -r requirements.txt
```
Comment (pafonta) [inline data_and_models/pipelines/ner/Dockerfile:45]: Shouldn't the two lines be inverted?

File: data_and_models/pipelines/sentence_embedding/Dockerfile [hunk 2]
```diff
@@ -33,7 +33,11 @@ RUN conda install -c carta mysqlclient
 ARG BBS_REVISION=v0.1.0
 ADD . /src
 WORKDIR /src
-RUN pip install git+file://$PWD@$BBS_REVISION
+RUN git checkout $BBS_REVISION
+# remove ruamel-yaml: https://github.com/pypa/pip/issues/5247#issuecomment-381550610
+RUN rm -rf /opt/conda/lib/python3.8/site-packages/ruamel*
+RUN pip install $PWD
+RUN pip install -r requirements.txt
```
Comment (pafonta) [inline data_and_models/pipelines/sentence_embedding/Dockerfile:40]: Same comment about the order.

File: data_and_models/pipelines/sentence_embedding/dvc.lock [hunk 3]
```diff
@@ -221,7 +221,7 @@ eval_biobert_nli_sts_cord19_v1:
     md5: f80bd92608012c65ede3417b10b0e731
     size: 1264
   - path: ../../metrics/sentence_embedding/biobert_nli_sts_cord19_v1.json
-    md5: c53426897361e4b4c66e58a78c85dad4
+    md5: c4371ca281d5f224332115f5a31ce9cd
```
Comment (pafonta) [inline data_and_models/pipelines/sentence_embedding/dvc.lock:224]: Have the performances of our production model changed then?

File: docs/source/whatsnew.rst [hunk 4]
```diff
@@ -37,8 +39,9 @@ Version 0.1.0
   depending on it, i.e. :code:`BSV` and :code:`Sent2VecModel`.
 - |Remove| the embedding model :code:`Universal Sentence Encoder`: (USE) and its
   dependencies (:code:`tensorflow` and :code:`tensorflow-hub`).
-- |Remove| :code:`BBS_BBG_poc` notebook (now hosted on https://github.com/BlueBrain/Search-Graph-Examples ) and
-  :code:`assets/` directory.
+- |Remove| :code:`BBS_BBG_poc` notebook (now hosted on
+  https://github.com/BlueBrain/Search-Graph-Examples ) and :code:`assets/`
```
Comment (pafonta) [inline docs/source/whatsnew.rst:43]:
```suggestion
  https://github.com/BlueBrain/Search-Graph-Examples) and :code:`assets/`
```

**Review by pafonta — COMMENTED** [review]
Thank you for the work!

I have followed the new testing instructions. I was able to reproduce the expected results (i.e. no output for `git diff` after `dvc repro -f`).

I don't approve yet because I have the 2 questions and the suggestion below.

**Review by jankrepl — APPROVED** [review]
LGTM! Thank you for the work!

File: data_and_models/pipelines/sentence_embedding/dvc.lock (hunk 3 above)
Comment (FrancescoCasalegno) [reply data_and_models/pipelines/sentence_embedding/dvc.lock:224]: I think the models predictions haven't changed—Kendall's τ and Spearman's ρ are the same. 
But probably `scipy` slightly changed the implementation of Pearson's r score— which is why we have some negligible difference (`~1e-7`) in the resulting Pearson's r value.

For a full reference, here you have all the changes:
```bash
bbs_292  [c4371ca281d5f224332115f5a31ce9cd] -- {"kendall_tau": 0.5778437302402729, "pearson_r": 0.7254631784119863, "spearman_rho": 0.7167140825467725}
master   [c53426897361e4b4c66e58a78c85dad4] -- {"kendall_tau": 0.5778437302402729, "pearson_r": 0.7254630892864342, "spearman_rho": 0.7167140825467725}
```

Comment (pafonta) [reply data_and_models/pipelines/sentence_embedding/dvc.lock:224]: Interesting. Thanks for the clarification.

File: data_and_models/pipelines/sentence_embedding/Dockerfile (hunk 2 above)
Comment (FrancescoCasalegno) [reply data_and_models/pipelines/sentence_embedding/Dockerfile]: good observation, fixed!

File: data_and_models/pipelines/ner/Dockerfile (hunk 1 above)
Comment (FrancescoCasalegno) [reply data_and_models/pipelines/ner/Dockerfile]: good observation, fixed!

Comment (pafonta) [reply data_and_models/pipelines/ner/Dockerfile]: Great!

**Review by pafonta — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**